src/cryptominisat.cpp

```
@@ -494,32 +494,34 @@ static bool actually_add_clauses_to_threads(CMSatPrivateData* data)
 
 DLL_PUBLIC void SATSolver::set_max_time(double max_time)
 {
-  for (size_t i = 0; i < data->solvers.size(); ++i) {
-    Solver& s = *data->solvers[i];
-    if (max_time >= 0) {
-      // the main loop in solver.cpp is checks `maxTime`
-      // against `cpuTime`, so we specify `s.conf.maxTime`
-      // as an offset from `cpuTime`.
-      s.conf.maxTime = cpuTime() + max_time;
-
-      //don't allow for overflow
-      if (s.conf.maxTime < max_time)
-          s.conf.maxTime = max_time;
-    }
+  if (max_time < 0) {
+    return;
+  }
+
+  const auto target_time = cpuTime() + max_time;
+  for (Solver* s : data->solvers) {
+    s->conf.maxTime = target_time;
   }
 }
 
 DLL_PUBLIC void SATSolver::set_max_confl(int64_t max_confl)
 {
-  for (size_t i = 0; i < data->solvers.size(); ++i) {
-    Solver& s = *data->solvers[i];
-    if (max_confl >= 0) {
-      s.conf.max_confl = s.get_stats().conflStats.numConflicts + max_confl;
+  if (max_confl < 0) {
+    return;
+  }
```

Again, please fail loudly, i.e. assert fail. Thanks!